Realm/ObjectStore/object_store.cpp

```diff
@@ -486,6 +515,65 @@ void ObjectStore::delete_data_for_object(Group *group, StringData object_type) {
     }
 }
 
+void ObjectStore::rename_property(Group *group, Schema& passed_schema, StringData object_type, StringData old_name, StringData new_name) {
+    Schema schema = schema_from_group(group);
```

Comment:
This could instead be:
```
    TableRef table = table_for_object_type(group, object_type);
    if (!table) {
        throw PropertyRenameMissingNewObjectTypeException(object_type);
    }
    ObjectSchema matching_schema(group, object_type);
```

which avoids the very heavyweight operation of reading the entire schema only to look at a single table.